### PR TITLE
Enable parallel transpile using API in broccoli-babel-transpiler

### PR DESCRIPTION
**Pull request by mikrostew**
Refactoring this plugin to run inline precompile in parallel, using the upcoming parallel API for `broccoli-babel-transpiler`: https://github.com/babel/broccoli-babel-transpiler/pull/114

### CODE REVIEW

**Comment by rwjblue** [comment]
Hmm. Why are changes required? If we need modifications here, I fear that this may not be a SemVer compatible change in the babel transpiler. What have I missed?

**Comment by mikrostew** [comment]
No changes are _required_ here for the change in `broccoli-babel-transpiler`. This will continue to work as-is, in series like it does now.

That change in `broccoli-babel-transpiler` will enable parallel babel transpilation for plugins that are able to take advantage of it. This PR is my attempt to take advantage of parallel transpilation for this plugin.

File: index.js [hunk 1]
```diff
@@ -52,44 +55,64 @@ module.exports = {
...
+            buildUsing: 'build',
+            params: {
+              templateCompilerPath,
+              parallelConfig
+            }
+          }
+        });
+      }
+      else {
+        _logger.debug('NOT using parallel API with broccoli-babel-transpiler');
```
Comment (stefanpenner) [inline index.js]: It would be nice if this function, and the one invoked while parallelized was the same function. I fear the risk of them diverging is high otherwise. And it would be nice to ensure the same tests could be used.

Comment (stefanpenner) [inline index.js]: if possible, can we print the name or paths (if available) of the plugins that prevented us from parallelizing. If its not possible, no worries.

Comment (mikrostew) [inline index.js]: done

Comment (mikrostew) [inline index.js]: done

File: config/ember-try.js [hunk 2]
```diff
@@ -1,5 +1,12 @@
 /*jshint node:true*/
+var os = require('os');
+var path = require('path');
+var username = require('username-sync')();
+var broccoliDiskCache = path.join(os.tmpdir(), username);
+
 module.exports = {
+  // clear broccoli async disk cache between scenarios for better isolation
+  command: 'rm -rf ' + broccoliDiskCache + '; ember test',
```
Comment (stefanpenner) [inline config/ember-try.js]: this is really surprising, why is this required?

File: index.js [hunk 3]
```diff
@@ -2,11 +2,14 @@
 'use strict';
 
 const path = require('path');
-const fs = require('fs');
 const hashForDep = require('hash-for-dep');
-const HTMLBarsInlinePrecompilePlugin = require('babel-plugin-htmlbars-inline-precompile');
+// const HTMLBarsInlinePrecompilePlugin = require('babel-plugin-htmlbars-inline-precompile');
```
Comment (stefanpenner) [inline index.js]: should we remove this comment?

File: lib/require-from-worker.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      const plugin = require(config.requireFile);
+      const buildInfo = plugin[config.buildUsing](config.params);
+
+      plugins.push(buildInfo.plugin);
+
+      if (typeof buildInfo.baseDir === 'function') {
+        const pluginHashForDep = hashForDep(buildInfo.baseDir());
+        cacheKeys.push(pluginHashForDep);
+      } else {
+        console.log('ember-cli-htmlbars-inline-precompile is opting out of caching due to an AST plugin that does not provide a caching strategy: `' + buildInfo.name + '`.');
```
Comment (stefanpenner) [inline lib/require-from-worker.js]: heimdall logger?

Comment (mikrostew) [inline lib/require-from-worker.js]: This was previously using `console.log` before I moved it - I'll update to use heimdall

File: index.js (hunk 3 above)
Comment (mikrostew) [inline index.js]: oops missed that, removed

File: config/ember-try.js (hunk 2 above)
Comment (mikrostew) [inline config/ember-try.js]: When running `ember try:each` 3 of the scenarios consistently failed - `ember-release`, `ember-beta`, and `ember-canary`. Running `ember try:one` for each of those scenarios individually worked fine though.

So I added this to clear the cache and force a fresh build every time, which worked to make the tests pass. But I need to go back and figure out the root cause of the failures.

**Comment by stefanpenner** [comment]
I would like @rwjblue to also sign-off, incase i missed something. Beyond that, I think due deligence is done, if issues arise we will address as they land.

**Comment by stefanpenner** [comment]
released as v1.0.0 🎉
